From c3f1b9a6dbaa2fc0b3ad4dee0f04c867b4eb3d5b Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Wed, 14 Jun 2023 00:48:21 +0200
Subject: [PATCH 465/469] pci: Workaround ITS timeouts on poweroff/reboot on
 Orange Pi 5 Plus

Call to pci_free_irq_vectors on PCIE2x ports causes ITS timeout messages
and delayed shutdown/reboot (by up to a minute).

The root cause will be elsewhere. This is not for upstream. Anyway,
there should be no harm from this, since we're shutting down anyway.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/pci/pcie/portdrv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/pcie/portdrv.c b/drivers/pci/pcie/portdrv.c
index 46fad0d813b2..1d35cb14e077 100644
--- a/drivers/pci/pcie/portdrv.c
+++ b/drivers/pci/pcie/portdrv.c
@@ -738,7 +738,7 @@ static void pcie_portdrv_shutdown(struct pci_dev *dev)
 		pm_runtime_dont_use_autosuspend(&dev->dev);
 	}
 
-	pcie_port_device_remove(dev);
+	device_for_each_child(&dev->dev, NULL, remove_iter);
 }
 
 static pci_ers_result_t pcie_portdrv_error_detected(struct pci_dev *dev,
-- 
2.34.1

